src/Compilers/VisualBasic/Portable/Binding/Binder_Expressions.vb

```diff
@@ -2096,9 +2096,9 @@ Namespace Microsoft.CodeAnalysis.VisualBasic
                 boundSecondArgWithConversions = MakeRValueAndIgnoreDiagnostics(boundSecondArg)
             End If
 
-            '  if there are still no errors check the original type of the first argument to be 
-            '       of a reference or nullable type, generate IllegalCondTypeInIIF otherwise
-            If Not hasErrors AndAlso Not (boundFirstArg.IsNothingLiteral OrElse boundFirstArg.Type.IsNullableType OrElse boundFirstArg.Type.IsReferenceType) Then
+            '  If there are still no errors check the original type of the first argument. If it's
+            '  a non-nullable value type, generate IllegalCondTypeInIIF
+            If Not hasErrors AndAlso Not boundFirstArg.IsNothingLiteral AndAlso boundFirstArg.Type.IsValueType AndAlso Not boundFirstArg.Type.IsNullableType() Then
```

Comment:
Should this new behavior be gated by language version and be documented in some list of VB16 features?
(I feel it should, since there is an accompanying spec change)

Reply:
Good catch, we need to do this for C# as well.

---
In reply to: [216848436](https://github.com/dotnet/roslyn/pull/29724#discussion_r216848436) [](ancestors = 216848436)